memfs/memory.go

```diff
@@ -154,6 +195,44 @@ func (fs *Memory) Dir(path string) billy.Filesystem {
 	}
 }
 
+func (fs *Memory) Symlink(target, link string) error {
+	if _, err := fs.Stat(link); err == nil {
+		return os.ErrExist
+	}
+
+	fullpath := clean(fs.Join(fs.base, link))
+	fs.links[fullpath] = internalLink{
+		Name:   link,
+		Target: clean(target),
+	}
+
+	return nil
+}
+
+func (fs *Memory) Readlink(link string) (string, error) {
+	fullpath := clean(fs.Join(fs.base, link))
+	if l, ok := fs.links[fullpath]; ok {
+		return l.Target, nil
+	}
+
+	return "", os.ErrNotExist
+}
+
+func (fs *Memory) resolvePath(path string) string {
```

Comment:
Move this function to the very beginning of the file, just after it is used for the first time.